pybind_interface/pybind_main.cpp

```diff
@@ -482,10 +482,37 @@ class SimulatorHelper {
                             unsigned>>& opsums_and_qubit_counts,
       bool is_noisy, const StateType& input_state) {
     auto helper = SimulatorHelper(options, is_noisy);
-    if (!helper.is_valid || !helper.simulate(input_state)) {
+    if (!helper.is_valid) {
       return {};
     }
-    return helper.get_expectation_value(opsums_and_qubit_counts);
+    if (!is_noisy) {
+      if (!helper.simulate(input_state)) {
+        return {};
+      }
+      return helper.get_expectation_value(opsums_and_qubit_counts);
+    }
+
+    // Aggregate expectation values for noisy circuits.
+    std::vector<std::complex<double>> results(
+      opsums_and_qubit_counts.size(), 0);
+    For aggregator(helper.num_threads);
+    auto add_evs = [&results](unsigned n, unsigned m, uint64_t i,
+                              const std::vector<std::complex<double>>& evs) {
+      results[i] += evs.at(i);
+    };
+    for (unsigned rep = 0; rep < helper.noisy_reps; ++rep) {
+      if (!helper.simulate(input_state)) {
+        return {};
+      }
+      auto evs = helper.get_expectation_value(opsums_and_qubit_counts);
+      aggregator.Run(evs.size(), add_evs, evs);
```

Comment:
It's probably better to use just the standard sequential `for` here and below. The parallel `for` will have advantage only for `opsums_and_qubit_counts` > 10000 or so.

Reply:
Done.